adding scripts for engineering

gasilva opened the pull request:
To implement three python scripts to give an example of powerful integration between jsbsim and python. For engineering purposes may be the most recommended tool.

All have graphics outputs but they can be changed to a table format.
The plot are more useful basically.

- catalog.py:
List properties of a script
This is good to know which variable to read and write

- runScriptLongFlight.py
Calculation required by aircraft icing enginering
Information required to calculate fuel consuption in leveled flight with no slats or flaps in order to provide data for mission profile analysis

it requires change in autopilot, trim-cruise script with autopilot and airborne.xml

- runJSBSimFuelMach.py
Calculation required by aircraft icing enginering
Find the optimum value for fuel consumption
For greatest distance - low AoA and high speed (Carson cruise)
For greatest time - high AoA and low speed (miminum power required)

The aoa.py appeared in the commit without adding it.
It appears to be the last version the in the github. I checkout the file before PR.

bcoconni commented:
Your PR is again polluted by unrelated changes.

Could you please run the following commands (assuming that you are using Linux of MacOSX) ?

First, make a backup copy of the scripts you want to push:
```bash
> mkdir save_for_PR
> cp examples/python/catalog.py save_for_PR/.
> cp examples/python/runJSBSimFuelMach.py save_for_PR/.
> cp examples/python/runScriptLongFlight.py save_for_PR/.
```
Then create a new branch `new_python_scripts` based on a pristine `master` branch from JSBSim:
```bash
> git branch new_python_scripts upstream/master
> git checkout new_python_scripts
> git pull upstream master
```
Restore your backup files in `examples/python`:
```bash
> cp save_for_PR/catalog.py examples/python/.
> cp save_for_PR/runJSBSimFuelMach.py examples/python/.
> cp save_for_PR/runScriptLongFlight.py examples/python/.
```
Create a new commit of your scripts **only**:
```bash
> git add examples/python/catalog.py examples/python/runJSBSimFuelMach.py examples/python/runScriptLongFlight.py
> git commit -m "adding scripts"
> git push origin new_python_scripts
```
Then create a new Pull Request in GitHub between your branch `gasilva/new_python_scripts` and `jsbsim/master`.

Now, you can delete the backup folder `save_for_PR` and switch back to your main branch:
```bash
> rm -fR save_for_PR/.
> git checkout master
```

seanmcleod commented:
@gasilva when you re-submit I'd suggest renaming the script files to match the filename convention we use in the `tests` directory, i.e. `RunJSBSimFuelMach.py` instead of `runJSBSimFuelMach.py`.

gasilva commented:
```
base) gasilva@kamino jsbsimNew % git add examples/python/Run*.py
(base) gasilva@kamino jsbsimNew % git add examples/python/Catalog.py
(base) gasilva@kamino jsbsimNew % git commit -m "adding scripts"    
On branch master
Your branch is ahead of 'origin/master' by 5 commits.
  (use "git push" to publish your local commits)

Changes not staged for commit:
	modified:   aircraft/global5000/global5000.xml
	modified:   examples/python/catalog.py
	modified:   examples/python/runJSBSimFuelMach.py
	modified:   examples/python/runScriptLongFlight.py

Untracked files:
	examples/python/cgVariation.py
	examples/python/speedsDesignManeuver.py
	examples/python/speedsTAS.py
	save_for_PR/

no changes added to commit
```

gasilva commented:
> Your PR is again polluted by unrelated changes.
> 
> Could you please run the following commands (assuming that you are using Linux or MacOSX) ?
> 
> First, make a backup copy of the scripts you want to push:
> 
> ```shell
> > mkdir save_for_PR
> > cp examples/python/catalog.py save_for_PR/.
> > cp examples/python/runJSBSimFuelMach.py save_for_PR/.
> > cp examples/python/runScriptLongFlight.py save_for_PR/.
> ```
> 
> Then create a new branch `new_python_scripts` based on a pristine `master` branch from JSBSim:
> 
> ```shell
> > git branch new_python_scripts upstream/master
> > git checkout new_python_scripts
> > git pull upstream master
> ```
> 
> Restore your backup files in `examples/python`:
> 
> ```shell
> > cp save_for_PR/catalog.py examples/python/.
> > cp save_for_PR/runJSBSimFuelMach.py examples/python/.
> > cp save_for_PR/runScriptLongFlight.py examples/python/.
> ```
> 
> Create a new commit of your scripts **only**:
> 
> ```shell
> > git add examples/python/catalog.py examples/python/runJSBSimFuelMach.py examples/python/runScriptLongFlight.py
> > git commit -m "adding scripts"
> > git push origin new_python_scripts
> ```
> 
> Then create a new Pull Request in GitHub between your branch `gasilva/new_python_scripts` and `jsbsim/master`.
> 
> Now, you can delete the backup folder `save_for_PR` and switch back to your main branch:
> 
> ```shell
> > rm -fR save_for_PR/.
> > git checkout master
> ```

The changes are six files:

- 3 python scripts
- airborne.xml
- global5000_ap.xml
- trim-cruise_ap.xml

the PR is listing 6 files, so I think it is not polluted.

bcoconni commented:
> The changes are six files:
>
> * 3 python scripts
> * airborne.xml
> * global5000_ap.xml
> * trim-cruise_ap.xml
>
> the PR is listing 6 files, so they are what supposed to be.

No I'm sorry but a PR must focus on **one single change**. 

I don't think it's a good idea to have too much interdependence between example scripts and the `global5000` aircraft otherwise we'll need to update the example scripts each time someone commit a change to the global 5000.

bcoconni commented:
Also I forgot to mention previously that you must supply the list of files to `git commit`:
```bash
> git commit -m "adding scripts" examples/python/runJSBSimFuelMach.py examples/python/runScriptLongFlight.py
```

gasilva commented:
> Also I forgot to mention previously that you must supply the list of files to `git commit`:
> 
> ```shell
> > git commit -m "adding scripts" examples/python/runJSBSimFuelMach.py examples/python/runScriptLongFlight.py
> ```

I had to git stash before the git pull
But now another error:

```shell
(base) gasilva@kamino jsbsimNew % git push origin new_python_scripts
Counting objects: 7, done.
Delta compression using up to 4 threads.
Compressing objects: 100% (7/7), done.
Writing objects: 100% (7/7), 4.62 KiB | 4.62 MiB/s, done.
Total 7 (delta 2), reused 0 (delta 0)
remote: Resolving deltas: 100% (2/2), completed with 2 local objects.
To https://github.com/gasilva/jsbsimNew.git
 ! [remote rejected]   new_python_scripts -> new_python_scripts (refusing to allow an OAuth App to create or update workflow `.github/workflows/cpp-python-build.yml` without `workflow` scope)
error: failed to push some refs to 'https://github.com/gasilva/jsbsimNew.git'
```

gasilva commented:
> > The changes are six files:
> > 
> > * 3 python scripts
> > * airborne.xml
> > * global5000_ap.xml
> > * trim-cruise_ap.xml
> > 
> > the PR is listing 6 files, so they are what supposed to be.
> 
> No I'm sorry but a PR must focus on **one single change**.
> 
> I don't think it's a good idea to have too much interdependence between example scripts and the `global5000` aircraft otherwise we'll need to update the example scripts each time someone commit a change to the global 5000.

The problem is that the files must also change..because now I have Mach hold...developed with @seanmcleod ... the trim-cruise, airborne and autopilot must adapt to it.

seanmcleod commented:
@gasilva I've been thinking of the best way to present these sorts of scripts to JSBSim users. I think rather than just having a bunch of python script files which initially a user will only look at based on the filename and then the documentation would be at best in code comments that we look at possibly rather linking to articles/posts from the JSBSim Wiki pages.

The posts will be able to better document the code/code snippets with images, formulas etc.

Here are a couple of examples I've created in the past.

https://seanmcleod.github.io/2021/06/inverted-trim/

https://seanmcleod.github.io/2018/02/landing-ground-effect-flare-jsbsim/

https://seanmcleod.github.io/2018/02/pitch-rate-damping-jsbsim-test/

The other option for the linked posts is to create a Jupyter notebook for each one and store it in a Github repo since Github will render a read-only version of the notebook.

For example:

https://github.com/seanmcleod/JupyterFTE/blob/master/Level%20Acceleration%20-%20Nigel%20Speedy.ipynb

https://github.com/seanmcleod/JupyterFTE/blob/master/Level%20Acceleration%20-%20Energy%20Height.ipynb

seanmcleod commented:
@gasilva I've added a new Wiki page - https://github.com/JSBSim-Team/jsbsim/wiki/Interesting-JSBSim-Examples

bcoconni commented:
@seanmcleod  Great ! :+1: It's a very worthy addition to our documentation.
